File: Exavision.Seasense.Shared/States/SiteState.cs
using System.Collections.Generic;

namespace Exavision.Seasense.Shared.States {
    public class SiteState : State {
        public List<MaterialState> Units { get; set; } = new List<MaterialState>();

        public List<CapabilityState> Capabilities { get; set; } = new List<CapabilityState>();


        public List<RecordingState> Recordings { get; set; } = new List<RecordingState>();

    }
}

File: Exavision.Seasense/Client/src/app/materials/states/site-state.ts
import { CapabilityState } from "./capability-state";
import { RecordingState } from "./recording-state";
import { UnitState } from "./unit-state";

export interface SiteState {
  $type: string;
  id: string;
  units: UnitState[];
  capabilities: CapabilityState[];
  recordings: RecordingState[];
}
